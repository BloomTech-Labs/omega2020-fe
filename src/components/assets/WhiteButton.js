import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { blue, grey } from '@material-ui/core/colors';
import { withStyles } from '@material-ui/core/styles';

const WhiteButton = (props) => {
  return (
    <ColorButton
      type={props.type}
      href={props.href}
      size='large'
      variant='outlined'
      color={blue.A700}
      onClick={props.click}
    >
      <Typography variant='caption'>{props.title}</Typography>
    </ColorButton>
  );
};

const ColorButton = withStyles(() => ({
  root: {
    paddingLeft: 30,
    paddingRight: 30,
    color: blue.A700,
    backgroundColor: grey[50],
    '&:hover': {
      backgroundColor: grey[300],
    },
    textTransform: 'none',
  },
}))(Button);

export default WhiteButton;
