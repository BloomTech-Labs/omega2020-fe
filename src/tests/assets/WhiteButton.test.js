import React from 'react';
import  { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';
import WhiteButton from '../../components/assets/WhiteButton';
import Adapter from 'enzyme-adapter-react-16';
import { Typography, Button } from '@material-ui/core';

configure({ adapter: new Adapter() });

describe('the white button content', () => {

    const mockInfo = {
        title: 'Play'
    }

    const ReactWrapper = mount(<WhiteButton mockInfo={mockInfo} />)

    it('snapshot the components', () => {
        const whiteButton = renderer.create(<WhiteButton />).toJSON()
        expect(whiteButton).toMatchSnapshot();
    })


    it('renders the mock info, and the length of the typography, and button', () => {
        expect(ReactWrapper.find(Typography)).toHaveLength(1);
        expect(ReactWrapper.find(Button)).toHaveLength(1);
        let wrapper = ReactWrapper.props().mockInfo.title
        expect(wrapper).toEqual(mockInfo.title)        
    })


    
})