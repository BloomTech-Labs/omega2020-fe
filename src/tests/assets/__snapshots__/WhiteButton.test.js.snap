// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the white button content snapshot the components 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-outlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <span
      className="MuiTypography-root MuiTypography-caption"
    />
  </span>
</button>
`;
